crate/ootr/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+#![deny(rust_2018_idioms, unused, unused_crate_dependencies, unused_import_braces, unused_qualifications, warnings)]
+#![forbid(unsafe_code)]
+
+use {
+    std::{
+        fs::{
+            self,
+            File,
+        },
+        io::{
+            Cursor,
+            prelude::*,
+        },
+    },
+    directories::ProjectDirs,
+    proc_macro::TokenStream,
+    pyo3::{
+        prelude::*,
+        types::PyDict,
+    },
+    quote::quote,
+    zip::ZipArchive,
+};
+
+const REPO_NAME: &str = "OoT-Randomizer";
+const LEAGUE_COMMIT_HASH: &str = "b670183e9aff520c20ac2ee65aa55e3740c5f4b4";
+const LEAGUE_VERSION: &str = "5.2.117 R-1";
+
+fn import<'p>(py: Python<'p>, module: &str) -> PyResult<&'p PyModule> {
+    let project_dirs = ProjectDirs::from("net", "Fenhl", "RSL").expect("missing home directory");
```

Comment:
is this a hardcoded path?

Reply:
Sort of. This uses the [directories](https://docs.rs/directories) crate to save things at the idiomatic location for each platform, e.g. `~\AppData\Local\Fenhl\RSL\cache` on Windows. This project uses that directory for the copies of the randomizer (latest/league), user settings (config.json, currently only whether to check for updates on launch), the plando and randomizer settings file that are saved here before running the randomizer, and the Python installer if the uses clicks the “Python not found — Install” button.